include/quic/utils.hpp

```diff
@@ -455,27 +466,52 @@ namespace oxen::quic
         explicit io_result(int errno_val) : error_code{errno_val} {}
 
         // Constructs an io_result with an ngtcp2 error value.
-        io_result(int err, ngtcp2_error_code_t) : error_code{err}, is_ngtcp2{true} {}
+        static io_result ngtcp2(int err)
```

Comment:
Minor: maybe renaming these to something like `ngtcp2_err` and `wsa_err` would be a bit clearer?

Reply:
They are static functions, so they always have to be prefixed with the `io_result::ngtcp(123)`